### PR TITLE
Min-utility best response from rideHailMaster.

### PR DESCRIPTION
**Pull request by jlaz**
Added option to choose between MIN_COST or MIN_UTILITY for choosing the best response from multiple RHM using the rideHailMaster.

### CODE REVIEW

File: src/main/scala/beam/agentsim/agents/ridehail/RideHailMaster.scala [hunk 1]
```diff
@@ -138,7 +143,62 @@ class RideHailMaster(
...
+    chosenProposal.get.alternativeType
+  }
+
+  private def proposalsToResponseAlternatives(
+    customer: Id[Person],
+    responses: IndexedSeq[RideHailResponse]
+  ): Map[RideHailResponse, Map[String, Double]] = {
+    val person = beamServices.matsimServices.getScenario.getPopulation.getPersons.get(customer)
+    val customerAttributes = person.getCustomAttributes.get(BEAM_ATTRIBUTES).asInstanceOf[AttributesOfIndividual]
+    responses.zipWithIndex.map { alt =>
```
Comment (dimaopen) [inline src/main/scala/beam/agentsim/agents/ridehail/RideHailMaster.scala]: I didn't find where the index is used

File: src/main/scala/beam/sim/config/BeamConfig.scala [hunk 2]
```diff
@@ -679,7 +680,9 @@ object BeamConfig {
                     else 90.0,
                   walk_intercept = if (c.hasPathOrNull("walk_intercept")) c.getDouble("walk_intercept") else 0.0,
                   walk_transit_intercept =
-                    if (c.hasPathOrNull("walk_transit_intercept")) c.getDouble("walk_transit_intercept") else 0.0
+                    if (c.hasPathOrNull("walk_transit_intercept")) c.getDouble("walk_transit_intercept") else 0.0,
+                  ride_hail_subscription =
+                    if (c.hasPathOrNull("ride_hail_subscription")) c.getDouble("ride_hail_subscription") else 0.0
```
Comment (dimaopen) [inline src/main/scala/beam/sim/config/BeamConfig.scala]: We shouldn't manually change `BeamConfig.scala`. We need to change `beam-template.conf` as described https://beam.readthedocs.io/en/latest/developers.html#configuration

**Review by dimaopen — CHANGES_REQUESTED** [review]

File: src/main/scala/beam/agentsim/agents/ridehail/RideHailMaster.scala (hunk 1 above)
Comment (jlaz) [reply src/main/scala/beam/agentsim/agents/ridehail/RideHailMaster.scala]: Agreed. I am removing .zipWithIndex and make corresponding edits.

File: src/main/scala/beam/sim/config/BeamConfig.scala (hunk 2 above)
Comment (jlaz) [reply src/main/scala/beam/sim/config/BeamConfig.scala]: Thanks. I have done this now

**Review by dimaopen — APPROVED** [review]

**Comment by nikolayilyin** [comment]
test!

**Pull request merged**